pkg/ast/spl/spl.peg

```diff
@@ -4276,12 +4276,45 @@ EarliestAndLatest <- CMD_EARLIEST EQUAL earliestTime:(AbsoluteTimestamp / Relati
     startEpoch, err = CalculateRelativeTime(earliestTimeModifier, currTime)
     if err != nil {
         return nil, err
+	}
+
+    for i := 0; i < len(earliestTimeModifier.ChainedOffsets) || i < len(earliestTimeModifier.ChainedSnaps); i++ {
+        var relativeTimeModifier ast.RelativeTimeModifier
+
+        if i < len(earliestTimeModifier.ChainedOffsets) {
+            relativeTimeModifier.RelativeTimeOffset = earliestTimeModifier.ChainedOffsets[i]
+        }
+
+        if i < len(earliestTimeModifier.ChainedSnaps) {
+            relativeTimeModifier.Snap = earliestTimeModifier.ChainedSnaps[i]
+        }
+
+        startEpoch, err = CalculateRelativeTime(ast.TimeModifier{RelativeTime: relativeTimeModifier}, time.UnixMilli(startEpoch))
+        if err != nil {
+            return nil, err
+        }
     }
```

Comment:
Extract this `for` loop into a function. Since it seems you have used the same one at other places also.

Reply:
Also probably a good idea to assign `len(*)` to variables and use them in the `for` loop. 

Also based on your `ChainedRelativeTimestamp` definition, the `ChainedOffsets` and `ChainedSnaps` should have the same length right? So may be you can remove the double check and simplify the code? 

Unless there can be a scenario where the length will not be same.

Reply:
Addressed it in the latest commit.